src/languages/rust/version.rs

```diff
@@ -0,0 +1,259 @@
+        semver::Version::parse(s).map(RustVersion)
+    }
+}
+
+/// `language_version` field of rust can be one of the following:
+/// `default`
+/// `system`
+/// `stable`
+/// `nightly`
+/// `beta`
+/// `1.70` or `1.70.0`
+/// `>= 1.70, < 1.72`
+/// `local/path/to/rust`
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub(crate) enum RustRequest {
+    Any,
+    Channel(String),
+    Major(u64),
+    MajorMinor(u64, u64),
+    MajorMinorPatch(u64, u64, u64),
+    Path(PathBuf),
+    Range(semver::VersionReq, String),
+}
+
+impl FromStr for RustRequest {
+    type Err = Error;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        if s.is_empty() {
+            return Ok(RustRequest::Any);
+        }
+
+        // Check for channel names
+        if s == "stable" || s == "nightly" || s == "beta" {
+            return Ok(RustRequest::Channel(s.to_string()));
+        }
+
+        // Try parsing as version numbers
+        Self::parse_version_numbers(s, s)
+            .or_else(|_| {
+                semver::VersionReq::parse(s)
+                    .map(|version_req| RustRequest::Range(version_req, s.into()))
+                    .map_err(|_| Error::InvalidVersion(s.to_string()))
+            })
+            .or_else(|_| {
+                let path = PathBuf::from(s);
+                if path.exists() {
+                    Ok(RustRequest::Path(path))
+                } else {
+                    Err(Error::InvalidVersion(s.to_string()))
+                }
+            })
+    }
+}
+
+impl RustRequest {
+    pub(crate) fn is_any(&self) -> bool {
+        matches!(self, RustRequest::Any)
+    }
+
+    fn parse_version_numbers(
+        version_str: &str,
+        original_request: &str,
+    ) -> Result<RustRequest, Error> {
+        let parts = try_into_u64_slice(version_str)
+            .map_err(|_| Error::InvalidVersion(original_request.to_string()))?;
+
+        match parts.as_slice() {
+            [major] => Ok(RustRequest::Major(*major)),
+            [major, minor] => Ok(RustRequest::MajorMinor(*major, *minor)),
+            [major, minor, patch] => Ok(RustRequest::MajorMinorPatch(*major, *minor, *patch)),
+            _ => Err(Error::InvalidVersion(original_request.to_string())),
+        }
+    }
+
+    pub(crate) fn satisfied_by(&self, install_info: &InstallInfo) -> bool {
+        let version = &install_info.language_version;
+
+        self.matches(
+            &RustVersion(version.clone()),
+            Some(install_info.toolchain.as_ref()),
+        )
+    }
+
+    pub(crate) fn matches(&self, version: &RustVersion, toolchain: Option<&Path>) -> bool {
+        match self {
+            RustRequest::Any => true,
+            RustRequest::Channel(_requested_channel) => {
+                // Cannot match channel names against specific version numbers
+                // e.g. request "stable" against version "1.70.0"
+                // TODO: Resolve channel by querying rustup or Rust release API
+                false
```

Comment:
Still TODO.

- The Node language dealt with LTS codenames via `EXTRA_KEY_LTS` [here](https://github.com/j178/prek/blob/1bb181ffa353581fb064c879b77e7aab15702439/src/languages/node/version.rs#L195)

Reply:
so this needs to store the channel and also treat an unspecified channel as "stable" to match what `resolve_version` will do